plugins/woocommerce/client/blocks/assets/js/blocks/featured-items/use-featured-item-status.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as coreDataStore } from '@wordpress/core-data';
+import { productsStore } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { BLOCK_NAMES } from './constants';
+
+interface UseFeaturedItemProps {
+	itemId: number | undefined;
+	itemType: string;
+}
+
+interface UseFeaturedItemReturnType {
+	status: string | null;
+	isDeleted: boolean | null;
+	isLoading: boolean;
+}
+
+export const useFeaturedItemStatus = ( {
+	itemId,
+	itemType,
+}: UseFeaturedItemProps ): UseFeaturedItemReturnType => {
+	return useSelect(
+		( selectFunc ) => {
+			if ( ! itemId ) {
+				return {
+					status: null,
+					isDeleted: true,
+					isLoading: false,
+				};
+			}
+
+			if ( itemType === BLOCK_NAMES.featuredProduct ) {
+				const { getProduct, hasFinishedResolution } =
```

Comment:
We've decided to transition from the `@woocommerce/data` package to the `@wordpress/core-data` package.  
Could you refactor the code accordingly and start using the `useProduct` hook? https://github.com/woocommerce/woocommerce/pull/60074/files#diff-b0dd86b8fdc84b4e6884bb0d17ab974f8da91364cba724a1218a15e7f2666e64R8
That would be great, thanks!